fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/federation/FederationExpect.kt

```diff
@@ -23,6 +24,14 @@ class FederationExpect
     val challenge: MessageGeneral
 ) : Data {
 
+  init {
+    MessageValidator.verify()
+        .isNotEmptyBase64(laoId, "lao_id")
+        .stringNotEmpty(serverAddress, "server_address")
```

Comment:
I am wondering if it would be a good idea to create a function in MessageValidator to check for server addresses pattern. It is specified in the protocol :

```
"server_address": {
            "type": "string",
            "pattern": "^(ws|wss):\/\/.*(:\\d{0,5})?\/.*$",
            "$comment": "public address of the remote organizer server"
        },
```

Reply:
For that point, the pattern might change as (if I remember well) some addresses didn't fit in the pattern. It will be added once the protocol is updated.